compiler/ast/enumtostr.nim

```diff
@@ -11,19 +11,21 @@ import
 
 
 proc genEnumToStrProc*(t: PType; info: TLineInfo; g: ModuleGraph; idgen: IdGenerator): PSym =
-  result = newSym(skProc, getIdent(g.cache, "$"), nextSymId idgen, t.owner, info)
+  # the compiler-generated enum-to-string procedure is treated as an
+  # instantiation of the generic ``mEnumToStr`` magic
+  let owner = getSysMagic(g, info, "$", mEnumToStr)
+  result = newSym(skProc, getIdent(g.cache, "$"), nextSymId idgen, owner, info)
```

Comment:
Yup, this feels a lot better in terms of the name of the proc and the actual implementation all matching up, with how everything else works.